api/document/js/commands.ts

```diff
@@ -69,3 +70,46 @@ export function makeSaveThenXml(api: Api) {
     window.location.assign(`${window.location.href}/akn`);
   };
 }
+
+const inLi = (pos: ResolvedPos) => (
+  pos.depth >= 3
+  && pos.node(pos.depth).type === schema.nodes.inline
+  && pos.node(pos.depth - 1).type === schema.nodes.para
+  && pos.node(pos.depth - 2).type === schema.nodes.listitem
+  && pos.node(pos.depth - 3).type === schema.nodes.list
+);
+const atEndOfLi = (pos: ResolvedPos) => (
+  pos.depth >= 2
+  && pos.pos === pos.end(pos.depth)
+  && pos.pos === pos.end(pos.depth - 1) - 1
+  && pos.pos === pos.end(pos.depth - 2) - 2
+);
+
+export function addListItem(state, dispatch?) {
```

Comment:
It might be useful, both for documentation and type-checking goodness, to annotate `state` as an [`EditorState`](https://prosemirror.net/docs/ref/#state.EditorState) here (and in other function declarations that take `state`).

It _might_ also be useful to define `dispatch` as `((tr: Transaction) => void) | null` (or make a type alias, since lots of functions take it) but up to you really.

Reply:
Good call, can do.